scripts/getArchive.py

```diff
@@ -61,7 +67,7 @@ def matches_filters(ent, args):
 
 
 def entity_trip(ent):
-    if "trip_update" in ent:
+    if "trip_update" in ent and ent["trip_update"] is not None:
```

Comment:
question: are we publishing null TripUpdates somewhere? I would expect the field to be missing rather than null where we don't provide it.

Reply:
We could create a rtr ticket for this. I'd rather not spend much more time on script itself though. We're better off spending that time on a livebook implementation. I have a work in progress one but it's not ready to be committed to this repository.

That said here's an example from the script of publishing an null TripUpdate within the VehiclePosition feed:

null TripUpdate within a VehiclePosition in rtr's enhanced vehicle position feed (*rtr_VehiclePositions_enhanced.json):
```
{
    "id": "1706048058_G-10186",
    "alert": None,
    "vehicle": {
        "position": {
            "speed": 5.516473599999999,
            "bearing": 45.0,
            "latitude": 42.32259,
            "longitude": -71.20525,
            "odometer": None,
        },
        "timestamp": 1706048058,
        "trip": {
            "start_time": "17:12:00",
            "start_date": "20240123",
            "trip_id": "60564884-20:15-BabckBackBayGovtGovtCtrKenmoreCDGovtCtrNorthStaGovtCtrNorthStaHeathNorthSusNorthMedfordNorthUnionSuspendD00",
            "route_id": "Green-D",
            "direction_id": 1,
            "schedule_relationship": "SCHEDULED",
            "revenue": True,
        },
        "stop_id": "70168",
        "vehicle": {
            "id": "G-10186",
            "label": "3677-3833",
            "consist": [{"label": "3677"}, {"label": "3833"}],
            "license_plate": None,
            "wheelchair_accessible": "NO_VALUE",
        },
        "multi_carriage_details": [
            {
                "id": None,
                "label": "3677",
                "occupancy_status": "NO_DATA_AVAILABLE",
                "occupancy_percentage": None,
                "carriage_sequence": 1,
            },
            {
                "id": None,
                "label": "3833",
                "occupancy_status": "NO_DATA_AVAILABLE",
                "occupancy_percentage": None,
                "carriage_sequence": 2,
            },
        ],
        "occupancy_status": None,
        "occupancy_percentage": None,
        "current_stop_sequence": 350,
        "current_status": "STOPPED_AT",
        "congestion_level": None,
    },
    "trip_update": None,
    "is_deleted": False,
    "shape": None,
}
```

`python -m pipenv run getArchive --datetime 2024-01-23T17:15:01 --feed subway_vehicle`